mail_restrict_send_button/static/src/js/chatter.js

```diff
@@ -0,0 +1,375 @@
+                default: false,
+            }),
+            isShowingAttachmentsLoading: attr({
+                default: false,
+            }),
+            /**
+             * Not a real field, used to trigger its compute method when one of the
+             * dependencies changes.
+             */
+            onThreadIdOrThreadModelChanged: attr({
+                compute: "_onThreadIdOrThreadModelChanged",
+                dependencies: ["threadId", "threadModel"],
+            }),
+            /**
+             * Not a real field, used to trigger its compute method when one of the
+             * dependencies changes.
+             */
+            onThreadIsLoadingAttachmentsChanged: attr({
+                compute: "_onThreadIsLoadingAttachmentsChanged",
+                dependencies: ["threadIsLoadingAttachments"],
+            }),
+            /**
+             * Determines the `mail.thread` that should be displayed by `this`.
+             */
+            thread: many2one("mail.thread"),
+            /**
+             * Determines the id of the thread that will be displayed by `this`.
+             */
+            threadId: attr(),
+            /**
+             * Serves as compute dependency.
+             */
+            threadIsLoadingAttachments: attr({
+                related: "thread.isLoadingAttachments",
+            }),
+            /**
+             * Serves as compute dependency.
+             */
+            threadIsTemporary: attr({
+                related: "thread.isTemporary",
+            }),
+            /**
+             * Determines the model of the thread that will be displayed by `this`.
+             */
+            threadModel: attr(),
+            /**
+             * States the `mail.thread_view` displaying `this.thread`.
+             */
+            threadView: one2one("mail.thread_view", {
+                related: "threadViewer.threadView",
+            }),
+            /**
+             * Determines the `mail.thread_viewer` managing the display of `this.thread`.
+             */
+            threadViewer: one2one("mail.thread_viewer", {
+                default: [["create"]],
+                inverse: "chatter",
+                isCausal: true,
+            }),
+        };
+
+        Chatter.modelName = "mail.chatter";
+
+        return Chatter;
+    };
+    container.prototype._insertFromProps = function (props) {
+        const values = Object.assign({}, props);
+        rpc.query({
+            model: "mail.followers",
+            method: "check_user_group",
+            args: [],
+        }).then((result) => {
+            values.isSendMessage = result;
+            if (values.threadId === undefined) {
+                values.threadId = clear();
+            }
+            if (!this.chatter) {
+                this.chatter = this.env.models["mail.chatter"].create(values);
+            } else {
+                this.chatter.update(values);
+            }
+        });
+    };
+});
```

Comment:
400 LoC to add a "isSendMessage" seems a bit too much.
is it possible to not repeat the original source code, and extend just adding the new attribute?